src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryUtilClasses.cs

```diff
@@ -0,0 +1,589 @@
+    internal sealed class ValueFixup
+    {
+        internal ValueFixupEnum _valueFixupEnum = ValueFixupEnum.Empty;
+        internal Array _arrayObj;
+        internal int[] _indexMap;
+        internal object _header = null;
+        internal object _memberObject;
+        internal static volatile MemberInfo _valueInfo;
+        internal ReadObjectInfo _objectInfo;
+        internal string _memberName;
+
+        internal ValueFixup(Array arrayObj, int[] indexMap)
+        {
+            _valueFixupEnum = ValueFixupEnum.Array;
+            _arrayObj = arrayObj;
+            _indexMap = indexMap;
+        }
+
+        internal ValueFixup(object memberObject, string memberName, ReadObjectInfo objectInfo)
+        {
+            _valueFixupEnum = ValueFixupEnum.Member;
+            _memberObject = memberObject;
+            _memberName = memberName;
+            _objectInfo = objectInfo;
+        }
+
+        internal void Fixup(ParseRecord record, ParseRecord parent)
+        {
+            object obj = record._PRnewObj;
+            switch (_valueFixupEnum)
+            {
+                case ValueFixupEnum.Array:
+                    _arrayObj.SetValue(obj, _indexMap);
+                    break;
+                case ValueFixupEnum.Header:
+                    Type type = typeof(Header);
+                    if (_valueInfo == null)
+                    {
+                        MemberInfo[] valueInfos = type.GetMember("Value");
+                        if (valueInfos.Length != 1)
+                        {
+                            throw new SerializationException(SR.Format(SR.Serialization_HeaderReflection, valueInfos.Length));
+                        }
+                        _valueInfo = valueInfos[0];
+                    }
+                    FormatterServices.SerializationSetValue(_valueInfo, _header, obj);
+                    break;
+                case ValueFixupEnum.Member:
+                    if (_objectInfo._isSi)
+                    {
+                        _objectInfo._objectManager.RecordDelayedFixup(parent._PRobjectId, _memberName, record._PRobjectId);
+                    }
+                    else
+                    {
+                        MemberInfo memberInfo = _objectInfo.GetMemberInfo(_memberName);
+                        if (memberInfo != null)
+                        {
+                            _objectInfo._objectManager.RecordFixup(parent._PRobjectId, memberInfo, record._PRobjectId);
+                        }
+                    }
+                    break;
+            }
+        }
+    }
+
+    // Class used to transmit Enums from the XML and Binary Formatter class to the ObjectWriter and ObjectReader class
+    internal sealed class InternalFE
+    {
+        internal FormatterTypeStyle _FEtypeFormat;
+        internal FormatterAssemblyStyle _FEassemblyFormat;
+        internal TypeFilterLevel _FEsecurityLevel;
+        internal InternalSerializerTypeE _FEserializerTypeEnum;
```

Comment:
The casing on the names here (and in NameInfo) seems off.  In Desktop they had no underscore, presenting as property names.  Now they are underscore-pascal; which I don't think is any part of our style rules.  Bad CodeFormatter edit?